### PR TITLE
Adding notes for how to do a release

### PR DESCRIPTION
**Pull request by arschles**
## What is the problem I am trying to address?

There were not reproducible notes on how to do an Athens release, so core maintainers had to just use "tribal knowledge" to remember how to do it.

## How is the fix applied?

I added docs on how to do everything.

## What GitHub issue(s) does this PR fix or close?

Fixes #

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -239,3 +239,55 @@ In our CI/CD pass, we use golint, so feel free to install and run it locally bef
...
+
+# For People Doing a Release
+
+This section is written primarily for maintainers, so we don't forget how to initiate and complete a release. If you're not a maintainer and you're curious how we release new versions of Athens, read on!
+
+Take a look at our [releases page](https://github.com/gomods/athens/releases). This is how we release official new builds of Athens. Almost all of the release process is automatic.
+
+## Code freeze
+
+The first step to a release is a code freeze. This is a period of time during which we don't merge anything to the `master` branch. The code in `master` is essentially a release candidate (we don't cut a new branch for RC's at the moment) to test. We'll push bugfixes to `master` only if needed.
```
Comment (marwan-at-work) [inline DEVELOPMENT.md]: I would specify the time, 1-2 weeks.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -239,3 +239,55 @@ In our CI/CD pass, we use golint, so feel free to install and run it locally bef
...
+
+## Release branch
+
+Right before we un-freeze the `master` branch, we will cut a new branch called `release-v1.X.Y`. Make sure the branch name starts with `release-v`, followed by the [semver](https://semver.org) release number.
```
Comment (marwan-at-work) [inline DEVELOPMENT.md]: I'd also put that you need to have permissions to create a new branch in `origin`, whether through the GitHub UI or through `git push origin release-v0.x.x`

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -239,3 +239,55 @@ In our CI/CD pass, we use golint, so feel free to install and run it locally bef
...
+
+### Note about the release number
+
+We follow the [semver](https://semver.org) convention for our release numbers. At the moment, we're not at `v1`, so release versions are `v0.x.y`. `x` is the `MINOR` number, and `y` is the `MAJOR` number. You'll need to decide which one to update.
```
Comment (twexler) [inline DEVELOPMENT.md]: This isn't right. `x` is MINOR, `y` is PATCH. MAJOR is `0` in this case (according to semver), we should stick to that.

Comment (arschles) [reply DEVELOPMENT.md]: Good catch, thanks @twexler

**Comment by arschles** [comment]
@marwan-at-work @twexler thanks for the reviews 😄  - I pushed some changes

**Pull request merged**